### PR TITLE
Add tests for project variables

### PR DESCRIPTION
**Pull request by vermakhushboo**
## What does this PR do?

Add missing tests for project variable endpoints

## Test Plan

(Write your test plan here. If you changed any code, please provide us with clear instructions on how you verified your changes work. Screenshots may also be helpful.)

## Related PRs and Issues

- (Related PR or issue)

## Checklist

- [ ] Have you read the [Contributing Guidelines on issues](https://github.com/appwrite/appwrite/blob/master/CONTRIBUTING.md)?
- [ ] If the PR includes a change to an API's metadata (desc, label, params, etc.), does it also include updated API specs and example docs?

### CODE REVIEW

File: tests/e2e/Services/Projects/ProjectsConsoleClientTest.php [hunk 1]
```diff
@@ -3793,4 +3793,273 @@ public function testDeleteProject(): array
...
+            'x-appwrite-mode' => 'admin',
+        ], $this->getHeaders()));
+
+        $this->assertEquals(404, $response['headers']['status-code']);
+
+        return $data;
+    }
+
+    /**
+     * @depends testGetVariable
```
Comment (Meldiron) [inline tests/e2e/Services/Projects/ProjectsConsoleClientTest.php:3922]: We can mark this as depending on testCreateVariable, mainly for clarity reasons. Logic-wise its fine both ways

File: tests/e2e/Services/Projects/ProjectsConsoleClientTest.php [hunk 2]
```diff
@@ -3793,4 +3793,273 @@ public function testDeleteProject(): array
...
+        $variable = $this->client->call(Client::METHOD_GET, '/project/variables/' . $data['variableId'], array_merge([
+            'content-type' => 'application/json',
+            'x-appwrite-project' => $data['projectId'],
+            'x-appwrite-mode' => 'admin',
+        ], $this->getHeaders()));
+
+        $this->assertEquals(200, $variable['headers']['status-code']);
+        $this->assertEquals("APP_TEST_UPDATE", $variable['body']['key']);
+        $this->assertEquals("TESTINGVALUEUPDATED", $variable['body']['value']);
+
```
Comment (Meldiron) [inline tests/e2e/Services/Projects/ProjectsConsoleClientTest.php:3952]: Let's also try to list variables here, and ensure value there is okay too

File: tests/e2e/Services/Projects/ProjectsConsoleClientTest.php [hunk 3]
```diff
@@ -3793,4 +3793,273 @@ public function testDeleteProject(): array
...
+        $variable = $this->client->call(Client::METHOD_GET, '/project/variables/' . $data['variableId'], array_merge([
+            'content-type' => 'application/json',
+            'x-appwrite-project' => $data['projectId'],
+            'x-appwrite-mode' => 'admin',
+        ], $this->getHeaders()));
+
+        $this->assertEquals(200, $variable['headers']['status-code']);
+        $this->assertEquals("APP_TEST_UPDATE_2", $variable['body']['key']);
+        $this->assertEquals("TESTINGVALUEUPDATED", $variable['body']['value']);
```
Comment (Meldiron) [inline tests/e2e/Services/Projects/ProjectsConsoleClientTest.php:3973]: I think we can remove second update&get test, I cant see edge case that would make it fail

File: tests/e2e/Services/Projects/ProjectsConsoleClientTest.php [hunk 4]
```diff
@@ -3793,4 +3793,273 @@ public function testDeleteProject(): array
...
+            'content-type' => 'application/json',
+            'x-appwrite-project' => $data['projectId'],
+            'x-appwrite-mode' => 'admin',
+        ], $this->getHeaders()), [
+            'key' => 'APP_TEST_UPDATE',
+            'value' => $longValue
+        ]);
+
+        $this->assertEquals(400, $response['headers']['status-code']);
+
```
Comment (Meldiron) [inline tests/e2e/Services/Projects/ProjectsConsoleClientTest.php:4020]: Let's include one 404 test - valid update request, but variable ID being non-existing

File: tests/e2e/Services/Projects/ProjectsConsoleClientTest.php (hunk 2 above)
Comment (Meldiron) [inline tests/e2e/Services/Projects/ProjectsConsoleClientTest.php:3952]: Let's listVariables and ensure value changed there  too

File: tests/e2e/Services/Projects/ProjectsConsoleClientTest.php [hunk 5]
```diff
@@ -3793,4 +3793,272 @@ public function testDeleteProject(): array
...
+         */
+
+        $response = $this->client->call(Client::METHOD_GET, '/project/variables', array_merge([
+            'content-type' => 'application/json',
+            'x-appwrite-project' => $data['projectId'],
+            'x-appwrite-mode' => 'admin',
+        ], $this->getHeaders()));
+
+        $this->assertEquals(200, $response['headers']['status-code']);
+        $this->assertEquals(1, sizeof($response['body']['variables']));
```
Comment (Meldiron) [inline tests/e2e/Services/Projects/ProjectsConsoleClientTest.php:3880]:
```suggestion
        $this->assertCount(1, $response['body']['variables']);
```

Comment (Meldiron) [reply tests/e2e/Services/Projects/ProjectsConsoleClientTest.php:3880]: Let's apply this in all places

**Review by Meldiron — APPROVED** [review]
